### DISCUSSION TITLE
Does Checkout Blocks allow Loqate or other address autofill extensions

### DISCUSSION THREAD

**Discussion by jnelson614** [woocommerce/woocommerce-gutenberg-products-block#11318] [Q&A - Extension Integrations and Customizations]
I am switching our store to use Checkout Blocks. We are currently using Loqate for Address validation and Auto fill, but it doesn't seem to work with Checkout blocks. Any tips, information, or ideas about this would be greatly appreciated!

-Jeff

**Comment by opr** [comment]
Hi there, thanks for reaching out. It looks like Loqate will need to add support for the Cart and Checkout blocks in their extension - we have the APIs in place for extensions to modify the address data and add validation errors etc.

I think your best bet would be to reach out to the vendor of this plugin and ask if they have plans to support WooCommerce Blocks.

There is already [some documentation available here](https://href.li/?https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/trunk/docs/third-party-developers/extensibility) that you could share with the developers of that extension now if you want to get this on their radar!

Until then, you can keep on top of the [status of the Cart & Checkout Blocks in this documentation](https://href.li/?https://docs.woocommerce.com/document/cart-checkout-blocks-support-status/).

I'll close this discussion as there's nothing actionable from our end right now, but thanks for bringing it to our attention!

**Reply by jnelson614** [reply]
Thanks @opr for the quick reply! Do you know of any address validation/auto fill that works with WooCommerce blocks?

Side note, and maybe I should open a different ticket for this, but with Google Autofill, the state always is selected wrong. I live in Massachusetts, so its saved as MA in my autofill settings. When applied to a WooCommerce checkout block form, it always selects Maine instead of Massachusetts.

**Reply by opr** [reply]
Hi @jnelson614 

> Do you know of any address validation/auto fill that works with WooCommerce blocks?

Sorry but I don't, I think there are a number of folks developing something similar (maybe without validation) in the [WC Community Slack](https://woocommerce.com/community-slack/) so it might be worth asking in there?

> Side note, and maybe I should open a different ticket for this, but with Google Autofill, the state always is selected wrong. I live in Massachusetts, so its saved as MA in my autofill settings. When applied to a WooCommerce checkout block form, it always selects Maine instead of Massachusetts.

Hmm, I guess because alphabetically the first state that matches those letters is Maine, but the actual value `MA` should probably be matched first. If you would, opening an issue would be helpful for us to track that. I think it's a valid bug and should be looked into. It might also require changes at the Gutenberg level, but we'll triage it and decide. Thanks a lot for raising this!

**Reply by jnelson614** [reply]
@opr Thanks! I opened a bug https://github.com/woocommerce/woocommerce-blocks/issues/11322
